tools/rack/rack.plugin/src/com/ge/research/rack/UploadNodegroupsHandler.java

```diff
@@ -74,6 +75,22 @@ public class UploadNodegroupsHandler extends AbstractHandler {
     private static final String NO_NODEGROUP_FILES =
             "Selected resource has no .json nodegroup files, try refreshing project";
 
+    // Debounce runs between all threads
+    private static boolean isRunning;
```

Comment:
better to make isRunning volatile, as upon any change to isRunning would be immediately reflected by a subsequent read

Reply:
I wanted to pull these commonalities into an interface - so far there are 2 handlers I know of that are long running but there may be more. Planning to dig deeper into eclipse plugin docs to see if this debounce functionality comes pre-packaged.